libazureinit/src/provision/config.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+use crate::error::Error;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use toml;
+use tracing;
+
+#[derive(Serialize, Deserialize, Debug)]
```

Comment:
Oh -- this is super cool. With this, we could eliminate the general azure-init.conf file and just merge in whatever we wanted to specify via 10-azure-init-ssh.conf. Is that what you were thinking?

Reply:
Basically, although I would not hard code a path. Instead, we can add a parameter to the command line so a user can provide `--config` or set the `AZURE_INIT_CONFIG` environment variable to a path where the configuration lives. Having a default of `/etc/azure-init.conf` or `/etc/azure-init/cli.conf` or something would be okay, but I don't think it's really necessary since we're not dictating where the config lives anywhere else, and we can provide reasonable defaults.

The distribution can install the config wherever it wants and configure the systemd unit (or whatever init system they use) to include an environment variable pointing to the config.